### PR TITLE
Remove extra copy of defer function and copy flow-typed version from debugger

### PR DESCRIPTION
**Pull request by wldcordeiro**
The defer function existed in two places and was identical, we also had a copy of it in debugger that was identical save for the flow typing that was added.

### CODE REVIEW

File: packages/devtools-launchpad/src/utils/defer.js [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
+declare var Reject: (result: any) => void;
+
+export type Defer = {
+  resolve: Resolve,
+  reject: Reject,
+  promise: Promise<any>
+};
+
+function defer(): Defer {
+  let resolve: Resolve; // eslint-disable-line no-unused-vars
```
Comment (tromey) [inline packages/devtools-launchpad/src/utils/defer.js]: Is this `eslint-disable` really needed?  It seems to me that `resolve` and `reject` are actually used.

**Comment by jasonLaster** [comment]
Lets just nuke defer in client-adapters. It's not used.

As a general point, I don't know how i feel about exporting modules from launchpad. Rules for me are
1. launchpad bootstrap
2. shims like context menu...

not sure about what else yet

**Comment by wldcordeiro** [comment]
@jasonLaster `devtools-utils`? 😄

**Comment by jasonLaster** [comment]
@wldcordeiro that sounds like devtools-modules

File: packages/devtools-launchpad/src/utils/defer.js (hunk 1 above)
Comment (jasonLaster) [inline packages/devtools-launchpad/src/utils/defer.js]: hmm, yeah - that can likely be removed

**Pull request merged**